### PR TITLE
DOC: Add examples to MultiIndex.slice_locs + note that index.slice requires

### PR DESCRIPTION
**Pull request by topper-123**
Clarify ``.slice_locs`` and set up examples for how to use the method.

Also, for ``pd.Index`` clarify that using ``.slice_locs`` requires that the index is monotonic.

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -3556,7 +3556,7 @@ def get_slice_bound(self, label, side, kind):
 
     def slice_locs(self, start=None, end=None, step=None, kind=None):
         """
-        Compute slice locations for input labels.
+        For an ordered index, compute slice locations for input labels.
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/base.py]: Index does not need to be ordered (it's only when there are duplicates that it has to be ordered I think). eg:

```
In [11]: pd.Index([1, 3, 4, 2]).slice_locs(1, 2)
Out[11]: (0, 4)
```

File: pandas/core/indexes/multi.py [hunk 2]
```diff
@@ -1944,7 +1946,32 @@ def slice_locs(self, start=None, end=None, step=None, kind=None):
...
         Notes
         -----
-        This function assumes that the data is sorted by the first level
+        This method only works if the MultiIndex is properly lex-sorted. So,
+        if only the first 2 levels of a 3-level MultiIndex  are lexsorted,
+        you can only pass two levels to ``.slice_locs``.
+
+        Examples
+        --------
+        mi = pd.MultiIndex.from_arrays([list('abbd'), list('deff')],
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/multi.py]: use 

```
>>>
...
```

notation for the examples

File: pandas/core/indexes/multi.py [hunk 3]
```diff
@@ -2015,6 +2042,8 @@ def get_loc(self, key, method=None):
         See also
         --------
         Index.get_loc : get_loc method for (single-level) index.
+        MultiIndex.slice_locs : Get slice location given startlabel(s) and
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/multi.py]: startlabel -> start label

**Review by jorisvandenbossche — COMMENTED** [review]
some comments!

File: pandas/core/indexes/base.py (hunk 1 above)
Comment (topper-123) [reply pandas/core/indexes/base.py]: Oh, thanks for catching that. Iøve reworded.

**Comment by topper-123** [comment]
Commit added doing the requested changes.

**Comment by jorisvandenbossche** [comment]
@topper-123 Thanks!

**Pull request merged**